scripts/jest/ts-preprocessor.js

```diff
@@ -28,7 +28,7 @@ function compile(content, contentFilename) {
       // the file path into backslashes on Windows.
       filename = path.normalize(filename);
       var reactRegex = new RegExp(
-        path.join('/', '(?:React|ReactDOM)(?:\.d)?\.ts$')
+        '\\' + path.join('/', '(?:React|ReactDOM)(?:\.d)?\.ts$')
```

Comment:
They help to normalize the path.But we need more work to construct a `RegExp` instance. 

![image](https://cloud.githubusercontent.com/assets/11155177/21867769/a70cfb50-d88a-11e6-9d35-f1885d46744c.png)

The backslash added here is to escape the leading separator, which is an special character in regular expression.

Oh,Wait.All of a sudden, I find the dots need to be escaped too.So I'll change it to:

```
var reactRegex = new RegExp(
  path.join('/', '(?:React|ReactDOM)(?:\.d)?\.ts$')
    .replace(/[\\.]/g, '\\$&')
);
```
Or more simple, if we don't mind matching files like XXXReact.d.ts:

```
var reactRegex = /(?:React|ReactDOM)(?:\.d)?\.ts$/;
```
@gaearon Which do you think would be better?